include/NovelRT/Atom.h

```diff
@@ -71,29 +72,33 @@ namespace NovelRT
             return _value;
         }
 
-        // TODO: These should be internal to NovelRT
-
-        static Atom GetNextEventHandlerId() noexcept;
-
-        static Atom GetNextFontSetId() noexcept;
-
-        static Atom GetNextTextureId() noexcept;
-
-        static Atom GetNextComponentTypeId() noexcept;
-
-        static Atom GetNextEntityId() noexcept;
-
-        static Atom GetNextSystemId() noexcept;
+        static Atom GetNextEcsPrimitiveInfoConfigurationId() noexcept;
+    };
 
-        static Atom GetNextEcsTextureId() noexcept;
+    class AtomFactory
+    {
+    private:
+        std::atomic_uintptr_t _currentValue;
+        bool _moved;
 
-        static Atom GetNextEcsVertexDataId() noexcept;
+    public:
+        AtomFactory() noexcept;
+        explicit AtomFactory(Atom startingValue) noexcept;
+        AtomFactory(const AtomFactory& other) noexcept;
+        AtomFactory(AtomFactory&& other) noexcept;
+        AtomFactory& operator=(const AtomFactory& other) noexcept;
+        AtomFactory& operator=(AtomFactory&& other) noexcept;
+        ~AtomFactory() = default;
 
-        static Atom GetNextEcsPrimitiveId() noexcept;
+        [[nodiscard]] Atom GetNext();
 
-        static Atom GetNextEcsGraphicsPipelineId() noexcept;
+        void SetToValue(Atom newValue);
+    };
 
-        static Atom GetNextEcsPrimitiveInfoConfigurationId() noexcept;
+    class AtomFactoryDatabase
+    {
+    public:
+        [[nodiscard]] static AtomFactory& GetFactory(const std::string& factoryName) noexcept;
     };
 
     class AtomHashFunction
```

Comment:
3 additional classes in one header file? o_O

Reply:
See previous comment